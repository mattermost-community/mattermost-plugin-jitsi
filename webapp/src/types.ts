import {ReactNode} from 'react';

import {GlobalState as ReduxGlobalState} from 'mattermost-redux/types/store';
import {Post} from 'mattermost-redux/types/posts';

export type Config = {
    TeammateNameDisplay?: string
}

export type GlobalState = ReduxGlobalState & {
    'plugins-jitsi': {
        openMeeting: Post | null,
        openMeetingJwt: string | null,
        config: {
            embedded?: boolean,
            /* eslint-disable camelcase */
            naming_scheme?: 'ask' | 'words' | 'mattermost' | 'uuid',
            use_jaas?: boolean
            show_prejoin_page: boolean
            /* eslint-enable camelcase */
        }
    }
}

export type InputElementType = HTMLInputElement | HTMLTextAreaElement;

export type RadioOptionsType = {
    value: string;
    checked: boolean;
    label: string | ReactNode;
}

export enum InputTypes {
    Text = 'text',
    Number = 'number',
    TextArea = 'textarea',
}

export type plugin = 'plugins-jitsi'
