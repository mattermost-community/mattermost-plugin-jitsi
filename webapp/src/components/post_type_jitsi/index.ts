import {connect} from 'react-redux';
import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';

import {getBool, getTheme} from 'mattermost-redux/selectors/entities/preferences';
import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
import {GenericAction, ActionFunc, ActionResult} from 'mattermost-redux/types/actions';
import {Post} from 'mattermost-redux/types/posts';

import {GlobalState} from '../../types';
import {displayUsernameForUser} from '../../utils/user_utils';
import {enrichMeetingJwt, openJitsiMeeting, setUserStatus} from '../../actions';
import {PostTypeJitsi} from './post_type_jitsi';

type OwnProps = {
    post: Post,
}

function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
    const post = ownProps.post;
    const creator = state.entities.users.profiles[post.user_id];
    const config = state['plugins-jitsi'].config;

    return {
        ...ownProps,
        currentUser: getCurrentUser(state),
        theme: getTheme(state),
        creatorName: displayUsernameForUser(creator, state.entities.general.config),
        useMilitaryTime: getBool(state, 'display_settings', 'use_military_time', false),
        meetingEmbedded: Boolean(config.embedded)
    };
}

type Actions = {
    enrichMeetingJwt: (jwt: string) => Promise<ActionResult>,
    openJitsiMeeting: (post: Post | null, jwt: string | null) => ActionResult,
    setUserStatus: (userId: string, status: string) => Promise<ActionResult>,
}

function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
    return {
        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc>, Actions>({
            enrichMeetingJwt,
            openJitsiMeeting,
            setUserStatus
        }, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(PostTypeJitsi);
