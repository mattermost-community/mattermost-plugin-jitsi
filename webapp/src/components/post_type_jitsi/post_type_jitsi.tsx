import * as React from 'react';
import {FormattedMessage} from 'react-intl';

import {Post} from 'mattermost-redux/types/posts';
import {Theme} from 'mattermost-redux/types/preferences';
import {ActionResult} from 'mattermost-redux/types/actions';

import Svgs from '../../constants/svgs';

import {makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';

export type Props = {
    post?: Post,
    theme: Theme,
    creatorName: string,
    useMilitaryTime: boolean,
    meetingEmbedded: boolean,
    actions: {
        enrichMeetingJwt: (jwt: string) => Promise<ActionResult>,
        openJitsiMeeting: (post: Post | null, jwt: string | null) => ActionResult,
    }
}

type State = {
    meetingJwt?: string,
}

export class PostTypeJitsi extends React.PureComponent<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {};
    }

    componentDidMount() {
        const {post} = this.props;
        if (post && post.props.jwt_meeting) {
            this.props.actions.enrichMeetingJwt(post.props.meeting_jwt).then((response: any) => {
                if (response.data) {
                    this.setState({meetingJwt: response.data.jwt});
                }
            });
        }
    }

    openJitsiMeeting = (e: React.MouseEvent) => {
        if (this.props.meetingEmbedded) {
            e.preventDefault();
            if (this.props.post) {
                this.props.actions.openJitsiMeeting(this.props.post, this.state.meetingJwt || this.props.post.props.meeting_jwt || null);
            }
        } else if (this.state.meetingJwt) {
            e.preventDefault();
            if (this.props.post) {
                const props = this.props.post.props;
                let meetingLink = props.meeting_link + '?jwt=' + (this.state.meetingJwt);
                meetingLink += `#config.callDisplayName="${props.meeting_topic || props.default_meeting_topic}"`;
                window.open(meetingLink, '_blank');
            }
        }
    }

    renderUntilDate = (post: Post, style: any): React.ReactNode => {
        const props = post.props;

        if (props.jwt_meeting) {
            const date = new Date(props.jwt_meeting_valid_until * 1000);
            let dateStr = props.jwt_meeting_valid_until;
            if (!isNaN(date.getTime())) {
                dateStr = date.toString();
            }
            return (
                <div style={style.validUntil}>
                    <FormattedMessage
                        id='jitsi.link-valid-until'
                        defaultMessage=' Meeting link valid until: '
                    />
                    <b>{dateStr}</b>
                </div>
            );
        }
        return null;
    }

    render() {
        const style = getStyle(this.props.theme);
        const post = this.props.post;
        if (!post) {
            return null;
        }

        const props = post.props;

        let meetingLink = props.meeting_link;
        if (props.jwt_meeting) {
            meetingLink += '?jwt=' + (props.meeting_jwt);
        }
        meetingLink += `#config.callDisplayName="${props.meeting_topic || props.default_meeting_topic}"`;

        const preText = (
            <FormattedMessage
                id='jitsi.creator-has-started-a-meeting'
                defaultMessage='{creator} has started a meeting'
                values={{creator: this.props.creatorName}}
            />
        );

        let subtitle = (
            <FormattedMessage
                id='jitsi.meeting-id'
                defaultMessage='Meeting ID: '
            />
        );
        if (props.meeting_personal) {
            subtitle = (
                <FormattedMessage
                    id='jitsi.personal-meeting-id'
                    defaultMessage='Personal Meeting ID (PMI): '
                />
            );
        }

        let title = (
            <FormattedMessage
                id='jitsi.default-title'
                defaultMessage='Jitsi Meeting'
            />
        );
        if (props.meeting_topic) {
            title = props.meeting_topic;
        }

        return (
            <div>
                {preText}
                <div style={style.attachment}>
                    <div style={style.content}>
                        <div style={style.container}>
                            <h1 style={style.title}>
                                {title}
                            </h1>
                            <span>
                                {subtitle}
                                <a
                                    target='_blank'
                                    rel='noopener noreferrer'
                                    onClick={this.openJitsiMeeting}
                                    href={meetingLink}
                                >
                                    {props.meeting_id}
                                </a>
                            </span>
                            <div>
                                <div style={style.body}>
                                    <div>
                                        <a
                                            className='btn btn-lg btn-primary'
                                            style={style.button}
                                            target='_blank'
                                            rel='noopener noreferrer'
                                            onClick={this.openJitsiMeeting}
                                            href={meetingLink}
                                        >
                                            <i
                                                style={style.buttonIcon}
                                                dangerouslySetInnerHTML={{__html: Svgs.VIDEO_CAMERA_3}}
                                            />
                                            <FormattedMessage
                                                id='jitsi.join-meeting'
                                                defaultMessage='JOIN MEETING'
                                            />
                                        </a>
                                    </div>
                                    {this.renderUntilDate(post, style)}
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const getStyle = makeStyleFromTheme((theme) => {
    return {
        attachment: {
            marginLeft: '-20px',
            position: 'relative'
        },
        content: {
            borderRadius: '4px',
            borderStyle: 'solid',
            borderWidth: '1px',
            borderColor: '#BDBDBF',
            margin: '5px 0 5px 20px',
            padding: '2px 5px'
        },
        container: {
            borderLeftStyle: 'solid',
            borderLeftWidth: '4px',
            padding: '10px',
            borderLeftColor: '#89AECB'
        },
        body: {
            overflowX: 'auto',
            overflowY: 'hidden',
            paddingRight: '5px',
            width: '100%'
        },
        title: {
            fontSize: '16px',
            fontWeight: '600',
            height: '22px',
            lineHeight: '18px',
            margin: '5px 0 1px 0',
            padding: '0'
        },
        button: {
            fontFamily: 'Open Sans',
            fontSize: '12px',
            fontWeight: 'bold',
            letterSpacing: '1px',
            lineHeight: '19px',
            marginTop: '12px',
            borderRadius: '4px',
            color: theme.buttonColor
        },
        buttonIcon: {
            paddingRight: '8px',
            fill: theme.buttonColor
        },
        validUntil: {
            marginTop: '10px'
        }
    };
});
