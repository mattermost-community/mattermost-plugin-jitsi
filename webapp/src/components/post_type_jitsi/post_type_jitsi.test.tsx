import * as React from 'react';
import {jest, describe, expect, it} from '@jest/globals';
import {shallow} from 'enzyme';

import {Post} from 'mattermost-redux/types/posts';

import {PostTypeJitsi} from './post_type_jitsi';
import Constants from 'mattermost-redux/constants/general';

describe('PostTypeJitsi', () => {
    const basePost: Post = {
        id: 'test',
        create_at: 100,
        update_at: 100,
        edit_at: 100,
        delete_at: 100,
        message: 'test-message',
        is_pinned: false,
        user_id: 'test-user-id',
        channel_id: 'test-channel-id',
        root_id: '',
        parent_id: '',
        original_id: '',
        type: 'custom_jitsi',
        hashtags: '',
        pending_post_id: '',
        reply_count: 0,
        metadata: {
            embeds: [],
            emojis: [],
            files: [],
            images: {},
            reactions: []
        },
        props: {
            jwt_meeting_valid_until: 123,
            meeting_link: 'http://test-meeting-link/test',
            jwt_meeting: true,
            meeting_jwt: 'xxxxxxxxxxxx',
            meeting_topic: 'Test topic',
            meeting_id: 'test',
            meeting_personal: false
        }
    };

    const actions = {
        enrichMeetingJwt: jest.fn().mockImplementation(() => Promise.resolve({data: {jwt: 'test-enriched-jwt'}})),
        openJitsiMeeting: jest.fn(),
        setUserStatus: jest.fn().mockImplementation(() => Promise.resolve({data: {user_id: 'test-user-id', status: Constants.DND}}))
    };

    const theme = {
        buttonColor: '#fabada'
    };

    const defaultProps = {
        post: basePost,
        theme,
        creatorName: 'test',
        currentUser: {
            first_name: 'First',
            last_name: 'Last'
        },
        useMilitaryTime: false,
        meetingEmbedded: false,
        actions
    };

    it('should render null if the post type is null', () => {
        defaultProps.actions.enrichMeetingJwt.mockClear();
        const props = {...defaultProps};
        delete props.post;
        const wrapper = shallow(
            <PostTypeJitsi {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(defaultProps.actions.enrichMeetingJwt).not.toBeCalled();
    });

    it('should render a post if the post type is not null, and should try to enrich the token', () => {
        defaultProps.actions.enrichMeetingJwt.mockClear();
        const wrapper = shallow(
            <PostTypeJitsi {...defaultProps}/>
        );
        expect(defaultProps.actions.enrichMeetingJwt).toBeCalled();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a post without token if there is no jwt token, and shouldn\'t try to enrich the token', () => {
        defaultProps.actions.enrichMeetingJwt.mockClear();
        const props = {
            ...defaultProps,
            post: {
                ...defaultProps.post,
                props: {
                    ...defaultProps.post.props,
                    jwt_meeting: false
                }
            }
        };

        const wrapper = shallow(
            <PostTypeJitsi {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(defaultProps.actions.enrichMeetingJwt).not.toBeCalled();
    });

    it('should render the default topic if the topic is empty', () => {
        const props = {
            ...defaultProps,
            post: {
                ...defaultProps.post,
                props: {
                    ...defaultProps.post.props,
                    meeting_topic: null
                }
            }
        };

        const wrapper = shallow(
            <PostTypeJitsi {...props}/>
        );
        expect(wrapper.find('h1')).toMatchSnapshot();
    });

    it('should render the a different subtitle if the meeting is personal', () => {
        const props = {
            ...defaultProps,
            post: {
                ...defaultProps.post,
                props: {
                    ...defaultProps.post.props,
                    meeting_personal: true
                }
            }
        };

        const wrapper = shallow(
            <PostTypeJitsi {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should prevent the default link behavior and call the action to open jitsi if embedded is true', () => {
        defaultProps.actions.openJitsiMeeting.mockClear();
        const props = {
            ...defaultProps,
            meetingEmbedded: true
        };

        const wrapper = shallow(<PostTypeJitsi {...props}/>);
        const event = {preventDefault: jest.fn()};
        wrapper.find('a.btn-primary').simulate('click', event);
        expect(defaultProps.actions.openJitsiMeeting).toBeCalled();
        expect(event.preventDefault).toBeCalled();
    });

    it('should not prevent the default link behavior and should not call the action to open jitsi if embedded is false', () => {
        defaultProps.actions.openJitsiMeeting.mockClear();
        const props = {
            ...defaultProps,
            meetingEmbedded: false
        };

        const wrapper = shallow(<PostTypeJitsi {...props}/>);
        const event = {preventDefault: jest.fn()};
        wrapper.find('a.btn-primary').simulate('click', event);
        expect(defaultProps.actions.openJitsiMeeting).not.toBeCalled();
        expect(event.preventDefault).not.toBeCalled();
    });
});
