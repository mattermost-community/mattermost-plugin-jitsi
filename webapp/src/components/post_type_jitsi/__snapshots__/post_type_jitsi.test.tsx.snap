// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTypeJitsi should render a post if the post type is not null, and should try to enrich the token 1`] = `
<div>
  <FormattedMessage
    defaultMessage="{creator} has started a meeting"
    id="jitsi.creator-has-started-a-meeting"
    values={
      Object {
        "creator": "test",
      }
    }
  />
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#BDBDBF",
          "borderRadius": "4px",
          "borderStyle": "solid",
          "borderWidth": "1px",
          "margin": "5px 0 5px 20px",
          "padding": "2px 5px",
        }
      }
    >
      <div
        style={
          Object {
            "borderLeftColor": "#89AECB",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "4px",
            "padding": "10px",
          }
        }
      >
        <h1
          style={
            Object {
              "fontSize": "16px",
              "fontWeight": "600",
              "height": "22px",
              "lineHeight": "18px",
              "margin": "5px 0 1px 0",
              "padding": "0",
            }
          }
        >
          Test topic
        </h1>
        <span>
          <FormattedMessage
            defaultMessage="Meeting ID: "
            id="jitsi.meeting-id"
            values={Object {}}
          />
          <a
            href="http://test-meeting-link/test?jwt=xxxxxxxxxxxx#config.callDisplayName=\\"Test topic\\"&userInfo.displayName=\\"First Last\\""
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            test
          </a>
        </span>
        <div>
          <div
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
                "paddingRight": "5px",
                "width": "100%",
              }
            }
          >
            <div>
              <a
                className="btn btn-lg btn-primary"
                href="http://test-meeting-link/test?jwt=xxxxxxxxxxxx#config.callDisplayName=\\"Test topic\\"&userInfo.displayName=\\"First Last\\""
                onClick={[Function]}
                rel="noopener noreferrer"
                style={
                  Object {
                    "borderRadius": "4px",
                    "color": "#fabada",
                    "fontFamily": "Open Sans",
                    "fontSize": "12px",
                    "fontWeight": "bold",
                    "letterSpacing": "1px",
                    "lineHeight": "19px",
                    "marginTop": "12px",
                  }
                }
                target="_blank"
              >
                <i
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg width='19px' height='10px' viewBox='0 0 19 10' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'><title>icon video small</title><desc>Created with Sketch.</desc><defs><linearGradient x1='50%' y1='0%' x2='50%' y2='100%' id='linearGradient-1'><stop stop-color='#FFFFFF' offset='0%'></stop><stop stop-color='#F4F4F4' offset='100%'></stop></linearGradient></defs><g id='v2' stroke='none' stroke-width='1' fill='currentColor' fill-rule='evenodd'><g id='zoom-bot-&amp;-drop-down' transform='translate(-443.000000, -250.000000)'><g id='zoom-bot---share-meeting' transform='translate(340.000000, 98.000000)'><g id='Send-Button' transform='translate(83.000000, 136.000000)'><g id='icon-video-small' transform='translate(20.000000, 16.000000)'><path d='M1,0 L10,0 L10,0 C12.209139,-4.05812251e-16 14,1.790861 14,4 L14,9 L14,9 C14,9.55228475 13.5522847,10 13,10 L4,10 L4,10 C1.790861,10 2.705415e-16,8.209139 0,6 L0,1 L0,1 C-6.76353751e-17,0.44771525 0.44771525,1.01453063e-16 1,0 Z' id='Rectangle-4' fill='currentColor'></path><path d='M15.3667766,2.86236308 L17.3667766,1.22606168 L17.3667766,1.22606168 C17.794228,0.876342075 18.4242495,0.93935609 18.7739691,1.36680742 C18.9201398,1.54546732 19,1.76919482 19,2.00003079 L19,8.99999935 L19,8.99999935 C19,9.5522841 18.5522847,9.99999935 18,9.99999935 C17.769164,9.99999935 17.5454365,9.92013917 17.3667766,9.77396845 L15.3667766,8.13766705 L15.3667766,8.13766705 C15.1346326,7.94773826 15,7.66363761 15,7.36369795 L15,3.63633218 L15,3.63633218 C15,3.33639253 15.1346326,3.05229188 15.3667766,2.86236308 Z' id='Rectangle-5' fill='currentColor'></path></g></g></g></g></g></svg>",
                    }
                  }
                  style={
                    Object {
                      "fill": "#fabada",
                      "paddingRight": "8px",
                    }
                  }
                />
                <FormattedMessage
                  defaultMessage="JOIN MEETING"
                  id="jitsi.join-meeting"
                  values={Object {}}
                />
              </a>
            </div>
            <div
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <FormattedMessage
                defaultMessage=" Meeting link valid until: "
                id="jitsi.link-valid-until"
                values={Object {}}
              />
              <b>
                Thu Jan 01 1970 00:02:03 GMT+0000 (Coordinated Universal Time)
              </b>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostTypeJitsi should render a post without token if there is no jwt token, and shouldn't try to enrich the token 1`] = `
<div>
  <FormattedMessage
    defaultMessage="{creator} has started a meeting"
    id="jitsi.creator-has-started-a-meeting"
    values={
      Object {
        "creator": "test",
      }
    }
  />
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#BDBDBF",
          "borderRadius": "4px",
          "borderStyle": "solid",
          "borderWidth": "1px",
          "margin": "5px 0 5px 20px",
          "padding": "2px 5px",
        }
      }
    >
      <div
        style={
          Object {
            "borderLeftColor": "#89AECB",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "4px",
            "padding": "10px",
          }
        }
      >
        <h1
          style={
            Object {
              "fontSize": "16px",
              "fontWeight": "600",
              "height": "22px",
              "lineHeight": "18px",
              "margin": "5px 0 1px 0",
              "padding": "0",
            }
          }
        >
          Test topic
        </h1>
        <span>
          <FormattedMessage
            defaultMessage="Meeting ID: "
            id="jitsi.meeting-id"
            values={Object {}}
          />
          <a
            href="http://test-meeting-link/test#config.callDisplayName=\\"Test topic\\"&userInfo.displayName=\\"First Last\\""
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            test
          </a>
        </span>
        <div>
          <div
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
                "paddingRight": "5px",
                "width": "100%",
              }
            }
          >
            <div>
              <a
                className="btn btn-lg btn-primary"
                href="http://test-meeting-link/test#config.callDisplayName=\\"Test topic\\"&userInfo.displayName=\\"First Last\\""
                onClick={[Function]}
                rel="noopener noreferrer"
                style={
                  Object {
                    "borderRadius": "4px",
                    "color": "#fabada",
                    "fontFamily": "Open Sans",
                    "fontSize": "12px",
                    "fontWeight": "bold",
                    "letterSpacing": "1px",
                    "lineHeight": "19px",
                    "marginTop": "12px",
                  }
                }
                target="_blank"
              >
                <i
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg width='19px' height='10px' viewBox='0 0 19 10' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'><title>icon video small</title><desc>Created with Sketch.</desc><defs><linearGradient x1='50%' y1='0%' x2='50%' y2='100%' id='linearGradient-1'><stop stop-color='#FFFFFF' offset='0%'></stop><stop stop-color='#F4F4F4' offset='100%'></stop></linearGradient></defs><g id='v2' stroke='none' stroke-width='1' fill='currentColor' fill-rule='evenodd'><g id='zoom-bot-&amp;-drop-down' transform='translate(-443.000000, -250.000000)'><g id='zoom-bot---share-meeting' transform='translate(340.000000, 98.000000)'><g id='Send-Button' transform='translate(83.000000, 136.000000)'><g id='icon-video-small' transform='translate(20.000000, 16.000000)'><path d='M1,0 L10,0 L10,0 C12.209139,-4.05812251e-16 14,1.790861 14,4 L14,9 L14,9 C14,9.55228475 13.5522847,10 13,10 L4,10 L4,10 C1.790861,10 2.705415e-16,8.209139 0,6 L0,1 L0,1 C-6.76353751e-17,0.44771525 0.44771525,1.01453063e-16 1,0 Z' id='Rectangle-4' fill='currentColor'></path><path d='M15.3667766,2.86236308 L17.3667766,1.22606168 L17.3667766,1.22606168 C17.794228,0.876342075 18.4242495,0.93935609 18.7739691,1.36680742 C18.9201398,1.54546732 19,1.76919482 19,2.00003079 L19,8.99999935 L19,8.99999935 C19,9.5522841 18.5522847,9.99999935 18,9.99999935 C17.769164,9.99999935 17.5454365,9.92013917 17.3667766,9.77396845 L15.3667766,8.13766705 L15.3667766,8.13766705 C15.1346326,7.94773826 15,7.66363761 15,7.36369795 L15,3.63633218 L15,3.63633218 C15,3.33639253 15.1346326,3.05229188 15.3667766,2.86236308 Z' id='Rectangle-5' fill='currentColor'></path></g></g></g></g></g></svg>",
                    }
                  }
                  style={
                    Object {
                      "fill": "#fabada",
                      "paddingRight": "8px",
                    }
                  }
                />
                <FormattedMessage
                  defaultMessage="JOIN MEETING"
                  id="jitsi.join-meeting"
                  values={Object {}}
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostTypeJitsi should render null if the post type is null 1`] = `""`;

exports[`PostTypeJitsi should render the a different subtitle if the meeting is personal 1`] = `
<div>
  <FormattedMessage
    defaultMessage="{creator} has started a meeting"
    id="jitsi.creator-has-started-a-meeting"
    values={
      Object {
        "creator": "test",
      }
    }
  />
  <div
    style={
      Object {
        "marginLeft": "-20px",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#BDBDBF",
          "borderRadius": "4px",
          "borderStyle": "solid",
          "borderWidth": "1px",
          "margin": "5px 0 5px 20px",
          "padding": "2px 5px",
        }
      }
    >
      <div
        style={
          Object {
            "borderLeftColor": "#89AECB",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "4px",
            "padding": "10px",
          }
        }
      >
        <h1
          style={
            Object {
              "fontSize": "16px",
              "fontWeight": "600",
              "height": "22px",
              "lineHeight": "18px",
              "margin": "5px 0 1px 0",
              "padding": "0",
            }
          }
        >
          Test topic
        </h1>
        <span>
          <FormattedMessage
            defaultMessage="Personal Meeting ID (PMI): "
            id="jitsi.personal-meeting-id"
            values={Object {}}
          />
          <a
            href="http://test-meeting-link/test?jwt=xxxxxxxxxxxx#config.callDisplayName=\\"Test topic\\"&userInfo.displayName=\\"First Last\\""
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            test
          </a>
        </span>
        <div>
          <div
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
                "paddingRight": "5px",
                "width": "100%",
              }
            }
          >
            <div>
              <a
                className="btn btn-lg btn-primary"
                href="http://test-meeting-link/test?jwt=xxxxxxxxxxxx#config.callDisplayName=\\"Test topic\\"&userInfo.displayName=\\"First Last\\""
                onClick={[Function]}
                rel="noopener noreferrer"
                style={
                  Object {
                    "borderRadius": "4px",
                    "color": "#fabada",
                    "fontFamily": "Open Sans",
                    "fontSize": "12px",
                    "fontWeight": "bold",
                    "letterSpacing": "1px",
                    "lineHeight": "19px",
                    "marginTop": "12px",
                  }
                }
                target="_blank"
              >
                <i
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<svg width='19px' height='10px' viewBox='0 0 19 10' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'><title>icon video small</title><desc>Created with Sketch.</desc><defs><linearGradient x1='50%' y1='0%' x2='50%' y2='100%' id='linearGradient-1'><stop stop-color='#FFFFFF' offset='0%'></stop><stop stop-color='#F4F4F4' offset='100%'></stop></linearGradient></defs><g id='v2' stroke='none' stroke-width='1' fill='currentColor' fill-rule='evenodd'><g id='zoom-bot-&amp;-drop-down' transform='translate(-443.000000, -250.000000)'><g id='zoom-bot---share-meeting' transform='translate(340.000000, 98.000000)'><g id='Send-Button' transform='translate(83.000000, 136.000000)'><g id='icon-video-small' transform='translate(20.000000, 16.000000)'><path d='M1,0 L10,0 L10,0 C12.209139,-4.05812251e-16 14,1.790861 14,4 L14,9 L14,9 C14,9.55228475 13.5522847,10 13,10 L4,10 L4,10 C1.790861,10 2.705415e-16,8.209139 0,6 L0,1 L0,1 C-6.76353751e-17,0.44771525 0.44771525,1.01453063e-16 1,0 Z' id='Rectangle-4' fill='currentColor'></path><path d='M15.3667766,2.86236308 L17.3667766,1.22606168 L17.3667766,1.22606168 C17.794228,0.876342075 18.4242495,0.93935609 18.7739691,1.36680742 C18.9201398,1.54546732 19,1.76919482 19,2.00003079 L19,8.99999935 L19,8.99999935 C19,9.5522841 18.5522847,9.99999935 18,9.99999935 C17.769164,9.99999935 17.5454365,9.92013917 17.3667766,9.77396845 L15.3667766,8.13766705 L15.3667766,8.13766705 C15.1346326,7.94773826 15,7.66363761 15,7.36369795 L15,3.63633218 L15,3.63633218 C15,3.33639253 15.1346326,3.05229188 15.3667766,2.86236308 Z' id='Rectangle-5' fill='currentColor'></path></g></g></g></g></g></svg>",
                    }
                  }
                  style={
                    Object {
                      "fill": "#fabada",
                      "paddingRight": "8px",
                    }
                  }
                />
                <FormattedMessage
                  defaultMessage="JOIN MEETING"
                  id="jitsi.join-meeting"
                  values={Object {}}
                />
              </a>
            </div>
            <div
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <FormattedMessage
                defaultMessage=" Meeting link valid until: "
                id="jitsi.link-valid-until"
                values={Object {}}
              />
              <b>
                Thu Jan 01 1970 00:02:03 GMT+0000 (Coordinated Universal Time)
              </b>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostTypeJitsi should render the default topic if the topic is empty 1`] = `
<h1
  style={
    Object {
      "fontSize": "16px",
      "fontWeight": "600",
      "height": "22px",
      "lineHeight": "18px",
      "margin": "5px 0 1px 0",
      "padding": "0",
    }
  }
>
  <FormattedMessage
    defaultMessage="Jitsi Meeting"
    id="jitsi.default-title"
    values={Object {}}
  />
</h1>
`;
